### DISCUSSION TITLE
Parallelisation of THM

### DISCUSSION THREAD

**Discussion by makeclean** [idaholab/moose#23276] [Q&A General] [answered]
Hi all, looking at parallelisation of THM calculations - looking at the news section within THM - https://mooseframework.inl.gov/modules/thermal_hydraulics/index.html - I see that it says ''The [Thermal hydraulics module](https://mooseframework.inl.gov/modules/thermal_hydraulics/index.html) can now run with distributed memory parallelism (MPI). Note that [HeatTransferFromHeatStructure1Phase](https://mooseframework.inl.gov/source/components/HeatTransferFromHeatStructure1Phase.html) currently requires a replicated mesh; see [issue](https://github.com/idaholab/moose/issues/20798)." 

The last time I tried to run my problem in parallel, which has HeatStructureFromFile3D and HeatTransferFromHeatStructure3D1Phase, I couldn't run in parallel which limits the scalability by using replicated mesh. I see that PR #20578 was merged in, but that looks like only part of the solution. If anyone could drop a few hints, I could have a crack and see what changes needed to be made? Its on my critical path so it makes me motivated :)

**Comment by GiudGiud** [comment, marked as answer]
@joshuahansel

**Reply by joshuahansel** [reply]
Yes, `HeatTransferFromHeatStructure3D1Phase` would have the same issue as `HeatTransferFromHeatStructure1Phase`. The issue you referenced: https://github.com/idaholab/moose/issues/20798, has everything that I know. I'll continue the discussion on that issue.
